src/xenia/kernel/xboxkrnl/xboxkrnl_memory.cc

```diff
@@ -424,7 +424,11 @@ DECLARE_XBOXKRNL_EXPORT2(MmQueryAddressProtect, kMemory, kImplemented,
 
 void MmSetAddressProtect(lpvoid_t base_address, dword_t region_size,
                          dword_t protect_bits) {
-  if (!protect_bits) {
+  constexpr uint32_t required_protect_bits =
+      X_PAGE_NOACCESS | X_PAGE_READONLY | X_PAGE_READWRITE |
```

Comment:
These sound mutually exclusive — are there real cases when games provide multiple of these at once? If there are none, the function should fail if `xe::bit_count(protect_bits & required_protect_bits) != 1`.

Reply:
Ohh great idea! 
I'm not aware of any game that was defining multiple bits at once.